src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -199,8 +199,24 @@ public URI getDeviceIdentityPrivateKeyURI() {
         return uriFromPossibleFileURIString(Coerce.toString(deviceConfiguration.getPrivateKeyFilePath()));
     }
 
+    /**
+     * Get URI for configured certificate-path.
+     * Configured certificate-path can be empty for PKCS11-type keys.
+     *
+     * @return URI of device-identity certificate
+     */
     public URI getDeviceIdentityCertificateURI() {
-        return uriFromPossibleFileURIString(Coerce.toString(deviceConfiguration.getCertificateFilePath()));
+        try {
+            String certPath = Coerce.toString(deviceConfiguration.getCertificateFilePath());
+            if (Utils.isNotEmpty(certPath)) {
+                return uriFromPossibleFileURIString(Coerce.toString(deviceConfiguration.getCertificateFilePath()));
+            }
+            // certificate path can be empty in case of PKCS11 keys
```

Comment:
The certificate must exist in the TPM, so why shouldn't we require that the URI to the cert in the TPM be specified?

Reply:
Because the field name is confusing. certificateFilePath
and java requires you to have same label on cert as private key anyway

Reply:
We require customers to use the same label when loading privateKey and certificate onto the TPM. Greengrass PKCS11 provider needs only this label to access these keys off of TPM, using KeyStore. Certificate-uri is redundant since we can use the same privateKey label, extracted from `privateKeyPath` uri. So customers may not specify `certificateFilePath` in config.

Here is the corresponding PR for PKCS11 provider: https://github.com/aws-greengrass/aws-greengrass-pkcs11-provider/pull/13